system/lib/libc/emscripten_memcpy.c

```diff
@@ -29,18 +20,26 @@ void *memcpy(void *dest, const void *src, size_t n) {
 
 #else
 
-void *memcpy(void *restrict dest, const void *restrict src, size_t n) {
+#ifndef STANDALONE_WASM
+// An external JS implementation that is efficient for very large copies, using
+// HEAPU8.set()
+void* emscripten_memcpy_big(void *restrict dest, const void *restrict src, size_t n) EM_IMPORT(emscripten_memcpy_big);
+#endif
+
+void *emscripten_builtin_memcpy(void *restrict dest, const void *restrict src, size_t n) {
   unsigned char *d = dest;
   const unsigned char *s = src;
 
   unsigned char *aligned_d_end;
   unsigned char *block_aligned_d_end;
   unsigned char *d_end;
 
+#ifndef STANDALONE_WASM
```

Comment:
This could be prefixed with `EMSCRIPTEN_*` like the others? (Some even do `__EMSCRIPTEN_X__` I see.)

Reply:
This is 100% internal so I'm not sure it matters?

Reply:
Yeah, it's just for internal consistency. Not that important.

Reply:
OK done.  I made them consistent.